src/main/java/io/vertx/core/file/impl/FileSystemImpl.java

```diff
@@ -602,6 +602,26 @@ public String createTempFileBlocking(String dir, String prefix, String suffix, S
     return createTempFileInternal(dir, prefix, suffix, perms).perform();
   }
 
+  static String getFileAccessErrorMessage(String verb, String path) {
+    return "Unable to " + verb + " file at path '" + path + "'";
+  }
+
+  static String getFolderAccessErrorMessage(String verb, String path) {
+    return "Unable to " + verb + " folder at path '" + path + "'";
+  }
+
+  static String getFileCopyErrorMessage(String from, String to) {
+    return getFileDualOperationErrorMessage("copy", from, to);
+  }
+
+  static String getFileMoveErrorMessage(String from, String to) {
+    return getFileDualOperationErrorMessage("move", from, to);
+  }
+
+  static String getFileDualOperationErrorMessage(String verb, String from, String to) {
```

Comment:
what means dual here ?

Reply:
It means that two paths are involved here. Operation ```delete file``` requires single file path, however operation ```copy file``` requires two paths (from and to).

I decided to extract function to cover this case. So logs will have both paths.